### PR TITLE
app-misc/pet: fix #650096

### PR DESCRIPTION
**Pull request by SpiderX**
1. Remove ARCHIVE_URI, move URI to SRC_URI
2. Use readme.gentoo-r1 and elog instead of einfo
3. Add slots to DEPEND
4. Rename pkg_postinstall to pkg_postinst (wrt #650096)

Closes: https://bugs.gentoo.org/650096
Package-Manager: Portage-2.3.19, Repoman-2.3.6

### CODE REVIEW

File: app-misc/pet/pet-0.2.4.ebuild [hunk 1]
```diff
@@ -20,13 +20,11 @@ EGO_VENDOR=( "github.com/BurntSushi/toml b26d9c308763d68093482582cea63d69be07a0f
```
Comment (mgorny) [inline app-misc/pet/pet-0.2.4.ebuild]: Please improve the summary of commit message.

1. Bug numbers require referencing external sources, please instead focus on what you're doing.
2. I don't think 650096 is somewhat 'most important' part of this commit. It looks like a major cleanup instead.
3. I think it would be better to do this in a series of split commits for every logical change.

File: app-misc/pet/pet-0.2.4.ebuild [hunk 2]
```diff
@@ -35,23 +33,29 @@ KEYWORDS="~amd64"
...
+DEPEND="dev-lang/go:=
+	dev-go/go-crypto:=
+	dev-go/go-net:=
+	dev-go/go-oauth2:=
+	dev-go/go-protobuf:=
+	dev-go/go-sys:="
 RDEPEND="${DEPEND}
 	zsh-completion? ( app-shells/zsh-completions )"
 
+DOC_CONTENTS="${CATEGORY}/${PN}: You should consider to
```
Comment (mgorny) [inline app-misc/pet/pet-0.2.4.ebuild]: 1. Portage already puts `cat/pn` above the message after finishing the builds.
2. To be honest, I don't think think a single sentence is really a reason to create a `README.gentoo` file.
3. Instead, you could use `has_version` to display it only when the dependency is not installed.
4. Also, there's `optfeature` thing in eutils that is kinda relevant, yet it's your choice if you want to use it.

**Comment by a17r** [comment]
ping

**Comment by SpiderX** [comment]
Split the previous commit into series of new ones with taking into account remarks from above.

File: app-misc/pet/pet-0.3.0.ebuild [hunk 3]
```diff
@@ -33,7 +35,8 @@ KEYWORDS="~amd64"
 IUSE="zsh-completion"
 
 # dev-go/toml doesn't provide sources
-DEPEND="dev-go/go-crypto:=
+DEPEND="dev-lang/go:=
```
Comment (mgorny) [inline app-misc/pet/pet-0.3.0.ebuild]: Doesn't golang-build handle depending on go?

File: app-misc/pet/pet-0.3.0.ebuild [hunk 4]
```diff
@@ -46,7 +49,7 @@ src_install() {
 
 	if use zsh-completion ; then
 		insinto /usr/share/zsh/site-functions
-		doins src/github.com/knqyf263/pet/misc/completions/zsh/_${PN}
+		doins src/github.com/knqyf263/pet/misc/completions/zsh/_"${PN}"
```
Comment (mgorny) [inline app-misc/pet/pet-0.3.0.ebuild]: This doesn't belong in this commit. That said, it's unnecessary to quote it here.

File: app-misc/pet/pet-0.3.0.ebuild [hunk 5]
```diff
@@ -4,19 +4,21 @@
 EAPI=6
 
 EGO_VENDOR=( "github.com/BurntSushi/toml b26d9c308763d68093482582cea63d69be07a0f0"
```
Comment (mgorny) [inline app-misc/pet/pet-0.3.0.ebuild]: Are you really fixing dependencies or did the dependencies require updating as part of bump? If the latter, then please merge it into the original commit doing the bump. It's good to split other changes into additional commits but changes that are necessary for the ebuild to work in the first place belong in the first commit.

File: app-misc/pet/pet-0.3.0.ebuild (hunk 3 above)
Comment (SpiderX) [reply app-misc/pet/pet-0.3.0.ebuild]: Yes, it inherits golang-base, which has a dependency on go.
dev-lang/go removed.

File: app-misc/pet/pet-0.3.0.ebuild (hunk 4 above)
Comment (SpiderX) [reply app-misc/pet/pet-0.3.0.ebuild]: A whole "${PN}" replaced by pet.

File: app-misc/pet/pet-0.3.0.ebuild (hunk 5 above)
Comment (SpiderX) [reply app-misc/pet/pet-0.3.0.ebuild]: Upstream changed dependencies - I am just following him in ebuild.
It always a good idea to have dependencies in upstream to be consistent with dependencies in ebuild. So, yes, dependencies require updating as a part of bump.

**Comment by mgorny** [comment]
Please use `git rebase -i` to squash the dependency update, and fixup it with go dep removal (no point in adding, then removing it).

**Comment by SpiderX** [comment]
Done.

**Comment by mgorny** [comment]
You still have that `#2` commit which is plain wrong.

**Comment by SpiderX** [comment]
I hope everything is right now.

**Comment by mgorny** [comment]
Thanks. I'll correct it to use GLEP 66 'Closes' tag and merge it if it works.
